### ISSUE TITLE
linter: False positive explicit-module-boundary-types

### ISSUE THREAD

**Issue by constantin-badan** [oxc-project/oxc#13209]
### What version of Oxlint are you using?

1.12

### What command did you run?

oxlint --quiet

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": ["typescript"],
  "rules": { "@typescript-eslint/explicit-module-boundary-types": "error" }
}
```

### What happened?

False positive for below
```tsx
function Test(): void {
  const _x = () => {};
}

export default Test;

```
